### PR TITLE
fix: error handling for auth0 source

### PR DESCRIPTION
**Pull request by Gauravudia**
## What are the changes introduced in this PR?

- Throwing userId not present error when userId is missing in all the request in a batch.

## What is the related Linear task?

Resolves INT-1140

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3038/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3038/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3038/test-report.html

File: src/v0/sources/auth0/transform.js [hunk 1]
```diff
@@ -73,7 +74,11 @@ function process(events) {
   if (!Array.isArray(events)) {
     eventList = events.logs || [events];
   }
-  return processEvents(eventList);
+  const responses = processEvents(eventList);
+  if (responses.length === 0) {
+    throw new InstrumentationError('UserId is not present. Aborting');
```
Comment (chandumlg) [inline src/v0/sources/auth0/transform.js:79]:
```suggestion
    throw new InstrumentationError('UserId is not present');
```

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3038/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3038/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3038/test-report.html

**Review by chandumlg — APPROVED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3038/test-report.html

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**